lib/actions/wallet.js

```diff
@@ -448,16 +453,60 @@ function sendTX(walletId, options) {
     assert(value, 'must enter value');
     assert(address, 'must enter recipient address');
 
-    const opts = {
+    let opts = {
       passphrase,
       rate,
       outputs: [{ value, address }],
     };
 
     try {
-      const response = await wallet.send(walletId, opts);
-      const txid = response.hash;
-      dispatch(setSendTXSuccess(walletId, accountId, txid, response));
+      // hardware wallet support
+      // create transaction on back end
+      // sign it using the hardware
+      if (options.hardware) {
+        const state = getState();
+        const chain = selectCurrentChain(state);
+        const network = selectNetwork(state);
+
+        const { node } = getClient();
+
+        // dont sign on back end
+        opts.sign = false;
+        const tx = await wallet.createTX(walletId, opts);
+
+        const paths = [];
+        const coins = [];
+        // iterate over tx inputs, building path
+        for (const input of tx.inputs) {
+          const info = await wallet.getKey(walletId, input.coin.address);
+          const path = templateBIP44Account(chain, network, info.account);
+          paths.push(`${path}/${info.branch}/${info.index}`);
+
+          // fetch raw tx
+          const inputTX = await node.getTX(input.prevout.hash);
```

Comment:
shouldn't you just be able to get the transaction from the wallet? We shouldn't need transaction indexing on for this. And if we do, we'll need this for the ez startup script